### PR TITLE
[Draft PR] Adding hint to the logger's error messages

### LINKED ISSUE #15143: Allow logging in `on_train_end`
## 🚀 Feature

I would like to be able to log the metrics of the best performing epoch at the end of training. This is especially useful for hyperparameter sweeps. `on_train_end` seems like a natural place for this.

### PR DESCRIPTION
**Pull request by Al3xDo**
## What does this PR do?

Fixes #15143 

## Before submitting

- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [ ] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing tests pass** locally with your changes?
- [ ] Did you list all the **breaking changes** introduced by this pull request?
- [ ] Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [ ] Is this pull request ready for review? (if not, please submit in draft mode)
- [ ] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

File: src/pytorch_lightning/trainer/connectors/logger_connector/fx_validator.py [hunk 1]
```diff
@@ -174,7 +174,7 @@ def check_logging(cls, fx_name: str) -> None:
             )
 
         if cls.functions[fx_name] is None:
-            raise MisconfigurationException(f"You can't `self.log()` inside `{fx_name}`.")
+            raise MisconfigurationException(f"You can alternatively log with `logger.experiment` inside `{fx_name}`.")
```
Comment (carmocca) [inline src/pytorch_lightning/trainer/connectors/logger_connector/fx_validator.py:177]: I would keep the old message, making it now

```
You can't `self.log()` inside `a_hook`. HINT: You can still log directly to the logger by using `self.logger.experiment`
```

And then you can revert the changes you had to do to the tests

**Review by carmocca — APPROVED** [review]
